### PR TITLE
Custom name visible

### PR DESCRIPTION
**Pull request by HSGamer**

### CODE REVIEW

File: src/main/java/plugily/projects/villagedefense/arena/Arena.java [hunk 1]
```diff
@@ -454,7 +454,10 @@ public void setWave(int wave) {
   }
 
   public void spawnVillager(Location location) {
-    addVillager(CreatureUtils.getCreatureInitializer().spawnVillager(location));
+    Villager villager = CreatureUtils.getCreatureInitializer().spawnVillager(location);
+    villager.setCustomNameVisible(plugin.getConfigPreferences().getOption(ConfigPreferences.Option.NAME_VISIBLE_VILLAGER));
+    villager.setCustomName(CreatureUtils.getRandomVillagerName());
```
Comment (montlikadani) [inline src/main/java/plugily/projects/villagedefense/arena/Arena.java:459]: Check if the custom name is visible before setting the name.

File: src/main/java/plugily/projects/villagedefense/creatures/CreatureUtils.java [hunk 2]
```diff
@@ -146,6 +147,10 @@ public static float getBabyZombieSpeed() {
     return villagerNames.clone();
   }
 
+  public static String getRandomVillagerName() {
+    return villagerNames[ThreadLocalRandom.current().nextInt(villagerNames.length)];
```
Comment (montlikadani) [inline src/main/java/plugily/projects/villagedefense/creatures/CreatureUtils.java:151]:
```suggestion
    return villagerNames[villagerNames.length == 1 ? 0 : ThreadLocalRandom.current().nextInt(villagerNames.length)];
```

File: src/main/java/plugily/projects/villagedefense/arena/Arena.java (hunk 1 above)
Comment (HSGamer) [reply src/main/java/plugily/projects/villagedefense/arena/Arena.java:459]: If it's false, the player can still see the name when he looks at the villager. It's intended.

Comment (montlikadani) [reply src/main/java/plugily/projects/villagedefense/arena/Arena.java:459]: I don't think so? If `setCustomNameVisible` was set to false, setting a new custom name to an entity will not display the name above the entity if the player is looking at.

Comment (HSGamer) [reply src/main/java/plugily/projects/villagedefense/arena/Arena.java:459]: I don't know what happened in legacy versions, but here is what I tested in 1.16.5

Comment (HSGamer) [reply src/main/java/plugily/projects/villagedefense/arena/Arena.java:459]: And this is Vanilla 1.16.5, no mods

**Comment by montlikadani** [comment]
idk if this works or not doesn't matter, setting `customNameVisible` to false will disable the custom name when looking at the entity,. For me this worked.

**Pull request merged**